src/spatialdata/transformations/_utils.py

```diff
@@ -72,13 +71,13 @@ def _set_transformations(e: SpatialElement, transformations: MappingToCoordinate
     raise TypeError(f"Unsupported type: {type(e)}")
 
 
-@_get_transformations.register(SpatialImage)
-def _(e: SpatialImage) -> Optional[MappingToCoordinateSystem_t]:
+@_get_transformations.register(DataArray)
+def _(e: DataArray) -> Optional[MappingToCoordinateSystem_t]:
```

Comment:
nitpicky but changing `Union` to `|` breaks singledispatch?

Reply:
Nitpicky change pushed. Also silenced some is_categorical_dtype deprecation warnings.